### PR TITLE
Enable static code analysis on CI and development

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR enables static code analysis checks both on CI and on dev machines, by adding the `make static-analysis` and `make static-analysis-docker` rules. Currently it should fail CI due to the issues(or false positives) raised on #562

### CODE REVIEW

File: src/version.h [hunk 1]
```diff
@@ -7,9 +7,9 @@
 
 #ifndef REDISTIMESERIES_MODULE_VERSION
 
-#define REDISTIMESERIES_VERSION_MAJOR 99
-#define REDISTIMESERIES_VERSION_MINOR 99
-#define REDISTIMESERIES_VERSION_PATCH 99
+#define REDISTIMESERIES_VERSION_MAJOR 1
```
Comment (danni-m) [inline src/version.h:10]: @filipecosta90 a bad merge from 1.4.5?

Comment (filipecosta90) [reply src/version.h:10]: your totally right. I saw it yesterday and had the idea that I force pushed the proper changes. Doing so :)

Comment (filipecosta90) [reply src/version.h]: @danni-m now fixed :+1:

**Comment by filipecosta90** [comment]
@danni-m now with the proper file changes, and failing as expected due to the issues(or false positives) raised on #562

**Pull request closed without merging**